raiden/network/pathfinding.py

```diff
@@ -1,12 +1,146 @@
-from typing import Dict, Union
-
 import requests
+
 from raiden.constants import DEFAULT_HTTP_REQUEST_TIMEOUT
+from raiden.exceptions import ServiceRequestFailed
+from raiden.utils import typing
+from raiden_contracts.utils.proofs import sign_one_to_n_iou
 
 
-def get_pfs_info(url: str) -> Union[Dict, bool]:
+def get_pfs_info(url: str) -> typing.Optional[typing.Dict]:
     try:
-        return requests.get('{}/api/v1/info'.format(url),
-                            timeout=DEFAULT_HTTP_REQUEST_TIMEOUT).json()
+        return requests.get(f'{url}/api/v1/info', timeout=DEFAULT_HTTP_REQUEST_TIMEOUT).json()
     except requests.exceptions.RequestException:
-        return False
+        return None
+
+
+def get_pfs_iou(
+        url: str,
+        token_network_address: typing.Union[typing.TokenNetworkAddress, typing.TokenNetworkID],
+        **kwargs,
+) -> typing.Optional[typing.Dict]:
+    try:
+        return requests.get(
+            f'{url}/api/v1/{token_network_address}/payment/iou',
+            data=kwargs,
+            timeout=DEFAULT_HTTP_REQUEST_TIMEOUT,
+        ).json().get('last_iou')
+    except (requests.exceptions.RequestException, ValueError) as e:
+        raise ServiceRequestFailed(str(e))
+
+
+def make_iou(
+        config: typing.Dict[str, typing.Any],
+        our_address: typing.Address,
+        privkey: str,
+        block_number: typing.BlockNumber,
+) -> typing.Dict:
+    expiration = block_number + config['pathfinding_iou_timeout']
+
+    iou = dict(
+        sender=our_address,
+        receiver=config['pathfinding_eth_address'],
+        amount=config['pathfinding_max_fee'],
+    )
+
+    iou.update(
+        expiration_block=expiration,
+        signature=sign_one_to_n_iou(privatekey=privkey, expiration=expiration, **iou),
+    )
+
+    return iou
+
+
+def update_iou(
+        iou: typing.Dict[str, typing.Any],
+        privkey: str,
+        added_amount: typing.TokenAmount = 0,
+        expiration_block: typing.Optional[typing.BlockNumber] = None,
+) -> typing.Dict[str, typing.Any]:
+
+    iou['amount'] += added_amount
+    if expiration_block:
+        iou['expiration_block'] = expiration_block
+
+    iou['signature'] = sign_one_to_n_iou(
+        privatekey=privkey,
+        expiration=iou['expiration_block'],
+        sender=iou['sender'],
+        receiver=iou['receiver'],
+        amount=iou['amount'],
+    )
+
+    return iou
+
+
+def create_current_iou(
+        config: typing.Dict[str, typing.Any],
+        token_network_address: typing.Union[typing.TokenNetworkAddress, typing.TokenNetworkID],
+        our_address: typing.Address,
+        privkey: str,
```

Comment:
`privkey` here should be typed as `bytes`

Reply:
Yes, thanks. I was confused as the `raiden_contracts` signing function expects a hex string.